pkg/storage/stores/tsdb/head_manager.go

```diff
@@ -131,10 +140,47 @@ func NewHeadManager(logger log.Logger, dir string, metrics *Metrics, tsdbManager
 
 	})
 
+	go m.loop()
+
 	return m
 }
 
+func (m *HeadManager) loop() {
+	m.loopWg.Add(1)
```

Comment:
Wondering why `wg.Add()` inside the spinning go routine itself (go `loop()`)?. I think `Add()` should be called before actually spinning the go routine.

https://pkg.go.dev/sync#WaitGroup.Add
```
Typically this means the calls to Add should execute before the statement creating the goroutine or other event to be waited for. If a WaitGroup is reused to wait for several independent sets of events, new Add calls must happen after all previous Wait calls have returned. See the WaitGroup example.
```

Reply:
agree, will move it out